File: Python/15-反转链表.py
# -*- coding:utf-8 -*-
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
class Solution:
    # 返回ListNode
    def ReverseList(self, pHead):
        # write code here
        rear = None
        p = pHead
        while p:
            q = p.next
            p.next = rear
            rear = p
            p = q
        return rear

File: C++/15-反转链表.cpp
/*
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};*/
class Solution {
public:
    ListNode* ReverseList(ListNode* pHead) {
        ListNode *rear = 0;
        ListNode* p = pHead, *q = 0;
        while(p){
            q = p->next;
            p->next = rear;
            rear = p;
            p = q;
        }
        p = q = 0;
        return rear;
    }
};
